hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashPolicyOzone.java

```diff
@@ -118,6 +120,25 @@ public Runnable getEmptier() throws IOException {
         emptierInterval);
   }
 
+  @Override
+  public boolean moveToTrash(Path path) throws IOException {
+    this.fs.getFileStatus(path);
+    Path trashRoot = this.fs.getTrashRoot(path);
+    Path trashCurrent = new Path(trashRoot, CURRENT);
+
+    String key = path.toUri().getPath();
+    String trashRootKey = trashRoot.toUri().getPath();
+
+    if (!OzoneFSUtils.isValidName(key)) {
+      throw new InvalidPathException("Invalid path Name " + key);
+    }
+
+    if (trashRootKey.startsWith(key) || key.startsWith(trashRootKey)) {
```

Comment:
I think only `key.startsWith(trashRootKey)` check would suffice. Do we need the OR condition?

Reply:
Thanks for asking to clarify.  Added comments in code to explain.  In process also improved conditional to satisfy an edge condition where key = .Tra (and the trash prefix is .Trash that would cause an unintentional action, by deleting and not moving to trash).
```
    // first condition tests when length key is <= length trash
    // and second when length key > length trash
    if ((key.contains(this.fs.TRASH_PREFIX)) && (trashRootKey.startsWith(key))
            || key.startsWith(trashRootKey)) {
```

Reply:
Thanks for the explaining it in the comment . Although , when length of  key < length of trashroot , the key would not be inside `.Trash` directory.
Suppose key = vol/buck/.Trash/user/randomKey ,the trashroot here would be vol/buck/.Trash 
So length of key is always greater than length of trashroot and first condition would not be required.

Reply:
Thanks @sadanand48, yes for what you described that is true.  The trashroot is derived from the bucket under the key,` trashroot`=`/vol/bucket/.Trash/username`.  One case where length `key` < length `trashroot` can be when `key` = `/vol/bucket/.Trash`; here `trashroot` is `/vol/bucket/.Trash/username`.

Reply:
Thanks for clarifying this @neils-dev . I forgot to consider the username in trashroot. It makes sense now . Thanks